### DISCUSSION TITLE
Using non-sans bold math

### DISCUSSION THREAD

**Discussion by legendre6891** [olsak/OpTeX#156] [Q&A] [answered]
Hi OpTeX users,

I am having trouble using non-sans bold math. My document is 
```
\_protected\_def\_mabf {\_inmath{\_bfvariables\_bfgreek\_bfGreek\_bfdigits}}

$\bf A$

\bye
```
(I copied the first line from the PDF manual.)

When I run this through OpTeX, I am getting the error
```
This is LuaTeX, Version 1.15.0 (TeX Live 2022)
 restricted system commands enabled.
(./luenberger-exercises.tex
This is OpTeX (Olsak's Plain TeX), version <1.13 Nov 2023>
! Undefined control sequence.
\_mabf ->\_inmath
                  {\_bfvariables \_bfgreek \_bfGreek \_bfdigits }
l.3 $\bf
       A$
```

Could someone give me a pointer? Much thanks

**Comment by legendre6891** [comment]
It seems like the problem is that the `\_inmath` macro is defined in `unimath-codes.opm`. Therefore, we need to load unicode font to make `\_inmath` available. 

For example, this works:
```
\fontfam[LMfonts]

\_protected\_def\_mabf {\_inmath{\_bfvariables\_bfgreek\_bfGreek\_bfdigits}}

$\bf A$

\bye
```

I'm curious if it's possible to do this without using `\fontfam[LMfonts]`?

**Comment by olsak** [comment, marked as answer]
Moving the definition of `\_inmath` from `unimath-codes` to the OpTeX core isn't sufficient, when you are using the example
```
\_protected\_def\_mabf {\_inmath{\_bfvariables\_bfgreek\_bfGreek\_bfdigits}}
```
This example is intended only for Unicode math, because the macros `\_bfvariables`, `\_bfgreek`, `\_bfGreek`, `\_bfdigits` are defined only when Unicode math is loaded. 

The non-Unicode math is pre-loaded in OpTeX only for testing purposes (for example for testing simple macros or for comparing non-Unicode math with Unicode math. The serious work needs Unicode math, IMHO.

If you insist on non-Unicode math in your document and you need original plain TeX `\bf`, then you have to do somewhat different work, because Unicode math macros and non-Unicode math macros are very different:

```
\_addto\_normalmath{\_loadmathfamily 8 cmbx }
\_normalmath

$\bf A$

\bye
```
The question remains: what to do with OpTeX so that other users don't have a problem with it. Maybe, this answer in this discussion is sufficient. Or I can add something into the documentation about this (but I don;t prefer it). The question+answer on this issue in tex.stackexchange should be good compromise, because such an answer is good searchable in the internet.

**Comment by legendre6891** [comment]
Thank you Petr; I read the documentation more thoroughly and understand/agree now that "`serious work needs Unicode math, IMHO`. Agree the current situation is sufficient.
